buildenvs/base.Dockerfile

```diff
@@ -2,114 +2,79 @@
+      make \
+      openssh-client \
+      patch \
+      python3 \
+      socat \
+      unzip \
+      uuid-runtime \
+      wget \
+      xz-utils \
+      ; \
     apt-get clean; \
     rm -Rf /var/cache/apt/*; \
-    rm -Rf /var/lib/apt/lists/*; \
-    kraft --log-type basic --log-level debug pkg update;
+    rm -Rf /var/lib/apt/lists/*
+
+RUN ln -s /usr/bin/cpp-12                                   /usr/bin/cc; \
+    ln -s /usr/bin/cpp-12                                   /usr/bin/cpp; \
+    ln -s /usr/bin/gcc-12                                   /usr/bin/gcc; \
+    ln -s /usr/bin/gcc-ar-12                                /usr/bin/gcc-ar; \
+    ln -s /usr/bin/gcc-nm-12                                /usr/bin/gcc-nm; \
+    ln -s /usr/bin/gcc-ranlib-12                            /usr/bin/gcc-ranlib; \
+    ln -s /usr/bin/gcov-12                                  /usr/bin/gcov; \
+    ln -s /usr/bin/gcov-dump-12                             /usr/bin/gcov-dump; \
+    ln -s /usr/bin/gcov-tool-12                             /usr/bin/gcov-tool; \
+    ln -s /usr/bin/lto-tool-12                              /usr/bin/lto-tool; \
+    ln -s /usr/bin/aarch64-linux-gnu-cpp-12                 /usr/bin/aarch64-linux-gnu-cpp; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcc-12                 /usr/bin/aarch64-linux-gnu-gcc; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcc-ar-12              /usr/bin/aarch64-linux-gnu-gcc-ar; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcc-nm-12              /usr/bin/aarch64-linux-gnu-gcc-nm; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcc-ranlib-12          /usr/bin/aarch64-linux-gnu-gcc-ranlib; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcov-12                /usr/bin/aarch64-linux-gnu-gcov; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcov-dump-12           /usr/bin/aarch64-linux-gnu-gcov-dump; \
+    ln -s /usr/bin/aarch64-linux-gnu-gcov-tool-12           /usr/bin/aarch64-linux-gnu-gcov-tool; \
+    ln -s /usr/bin/aarch64-linux-gnu-lto-tool-12            /usr/bin/aarch64-linux-gnu-lto-tool; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-cpp-12        /usr/bin/gcc-12-arm-linux-gnueabihf-cpp; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-12        /usr/bin/gcc-12-arm-linux-gnueabihf-gcc; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-ar-12     /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-ar; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-nm-12     /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-nm; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-ranlib-12 /usr/bin/gcc-12-arm-linux-gnueabihf-gcc-ranlib; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcov-12       /usr/bin/gcc-12-arm-linux-gnueabihf-gcov; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcov-dump-12  /usr/bin/gcc-12-arm-linux-gnueabihf-gcov-dump; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-gcov-tool-12  /usr/bin/gcc-12-arm-linux-gnueabihf-gcov-tool; \
+    ln -s /usr/bin/gcc-12-arm-linux-gnueabihf-lto-tool-12   /usr/bin/gcc-12-arm-linux-gnueabihf-lto-tool;
```

Comment:
Why is this needed if we install from packages?

Reply:
Unfortunately all programs are suffixed with `-12`. These are simply aliases to what is installed.